sdk/eventhub/Azure.Messaging.EventHubs/tests/Shared/ConnectionStringParser.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Azure.Messaging.EventHubs.Tests.Infrastructure
+{
+    /// <summary>
+    ///   General purpose parser that allows to retrieve a token in connection string.
+    /// </summary>
+    ///
+    internal static class ConnectionStringParser
+    {
+        /// <summary>
+        ///   It parses a connection string looking for the provided token name and returns its value.
+        /// </summary>
+        ///
+        /// <param name="connectionString">An Event Hubs namespace connection string</param>
+        /// <param name="tokenKey">The token to be looked for</param>
+        ///
+        /// <returns>The value of the token if found; otherwise, a <see cref="InvalidOperationException" /> is thrown.</returns>
+        ///
+        internal static string ParseTokenAndReturnValue(string connectionString, string tokenKey)
+        {
+            string token = ParseToken(connectionString, tokenKey);
+
+            if (string.IsNullOrEmpty(token))
+            {
+                throw new InvalidOperationException($"The { tokenKey } token was not found.");
+            }
+
+            string tokenValue = ParseTokenValue(token);
+
+            if (string.IsNullOrEmpty(tokenValue))
+            {
+                throw new InvalidOperationException($"The { tokenKey } token is not well-formed.");
+            }
+
+            return tokenValue;
+        }
+
+        /// <summary>
+        ///   It parses a connection string looking for the provided token and returns it.
+        /// </summary>
+        ///
+        /// <param name="connectionString">An Event Hubs namespace connection string</param>
+        /// <param name="tokenKey">The token key to be looked for</param>
+        ///
+        /// <returns>The token if found; otherwise, null.</returns>
+        ///
+        internal static string ParseToken(string connectionString, string tokenKey)
+        {
+            var regex = new Regex($"(?<token>{ tokenKey }=[^;]+(;|$))");
```

Comment:
Well.... you're a much smarter human than I with your mastery of mysterious RegEx patterns.   Kidding aside, I think this is a good general approach for the test space.   I'd love to see this migrate to the new Shared project testing area and replace my quick-n-dirty string parse.

I'd prefer to keep the raw string parsing for the client itself, as it should allocate less and perform trivially better than going RegEx given the simple forward-only semantics.

Reply:
> you're a much smarter human than I with your mastery of mysterious RegEx patterns.

You really want to fool me into believing you are human? 😛